Added sample csv parsing, custom comparer for response tables, and some documentation

zchaudhry11 opened the pull request:
I created a CSV parser that can parse the process spreadsheet that is being filled out and linked it to the ApplicationBehavior class. It seems to work fine from the testing I've done so far and I will work on linking it to the WebBrowser behavior once I have made sure that there are no edge cases that are causing problems. On a side note, the updater appears to be broken at the moment so I also included a sample csv file that I have been parsing.

PiMaker commented:
Your code looks really good! Only two things I have noticed, although I could do these on my own too, after merging:

* The special browser handling is at the right place, but the function that tests for browser behaviour should be passed directly into the dictionary. So don't call it and then just pass in the boolean, put the entire method in.
* And for the CSV file: How exactly did you get your example one? Is it a raw export? Because in my opionion it would be best for it to be, or at least be as close to one as possible while still being read in well enough. That would eliminate some work on my end whenever I decide to update the Spreadsheet being passed out.

zchaudhry11 commented:
Thanks for the swift response! I synced my repository and I believe I fixed the issue you had with the browser handling method.

As for the CSV file, I just went to the public spreadsheet listed in the readme and went to the Applications tab and then just used the Google Sheets download button to get the page in csv format. From there I opened it in Excel and changed it to be separated by semi-colons, and then I manually put it in the Roaming folder since the auto updater still doesn't seem to be working (It still maxes out all 60 API requests that can be made in an hour on launch even with the latest code.)
